.github/workflows/test.yml

```diff
@@ -56,15 +55,22 @@ jobs:
     - name: Install package dependencies
       run: mix deps.get
 
+    - name: Remove compiled application files
+      run: mix clean
+
+    - name: Compile dependencies
+      run: mix compile
```

Comment:
Why were these steps added? It's unclear from your commit message and any additional context would be useful for reviewers.

Reply:
Running tests already compiles all the dependencies as well as the package. Sometimes compilation fails, and this leads to the failing "Run unit tests" or "Check documentation quality" steps, but they are not really failing.
Here I've just separated the compilation step and moved id before the check.